core/node/node_framework/src/implementations/layers/settlement_layer_data.rs

```diff
@@ -175,12 +186,150 @@ impl WiringLayer for SettlementLayerData {
+            )
+            .await?
+        };
+
+        let l2_eth_client = get_l2_client(self.config.gateway_rpc_url, initial_sl_mode, chain_id)?;
+
+        let (client, bridgehub): (&dyn EthInterface, Address) = match initial_sl_mode {
+            SettlementMode::SettlesToL1 => (
+                &input.eth_client.0,
+                self.config
+                    .l1_chain_contracts
+                    .ecosystem_contracts
+                    .bridgehub_proxy_addr
+                    .unwrap(),
+            ),
+            SettlementMode::Gateway => (&l2_eth_client.as_ref().unwrap().0, L2_BRIDGEHUB_ADDRESS),
+        };
+
+        // There is no need to specify multicall3 for external node
+        let contracts =
+            load_settlement_layer_contracts(client, bridgehub, self.config.chain_id, None).await?;
+        let contracts = match contracts {
+            Some(contracts) => contracts,
+            None => match initial_sl_mode {
+                SettlementMode::SettlesToL1 => self.config.l1_chain_contracts.clone(),
+                SettlementMode::Gateway => {
+                    return Err(anyhow::anyhow!("No contacts deployed to contracts"))?
+                }
+            },
+        };
+
+        Ok(Output {
+            contracts: SettlementLayerContractsResource(contracts),
+            l1_contracts: L1ChainContractsResource(self.config.l1_chain_contracts),
+            l1_ecosystem_contracts: L1EcosystemContractsResource(self.config.l1_specific_contracts),
+            l2_contracts: L2ContractsResource(self.config.l2_contracts),
+            initial_settlement_mode: SettlementModeResource(initial_sl_mode),
+            sl_chain_id: SlChainIdResource(chain_id),
+            l2_eth_client,
+            eth_sender_config: None,
+        })
+    }
+}
+
+fn get_l2_client(
+    gateway_rpc_url: Option<SensitiveUrl>,
+    initial_sl_mode: SettlementMode,
+    chain_id: SLChainId,
+) -> anyhow::Result<Option<L2InterfaceResource>> {
+    Ok(gateway_rpc_url
+        .map(|url| Client::http(url).context("Client::new()"))
+        .transpose()?
+        .and_then(|builder| {
+            initial_sl_mode.is_gateway().then(|| {
+                Some(L2InterfaceResource(Box::new(
+                    builder
+                        .for_network(L2ChainId::new(chain_id.0).unwrap().into())
+                        .build(),
+                )))
+            })
+        })
+        .flatten())
+}
+
+fn adjust_eth_sender_config(
+    mut config: SenderConfig,
+    settlement_mode: SettlementMode,
+) -> SenderConfig {
+    if settlement_mode.is_gateway() {
+        config.max_aggregated_tx_gas = 4294967295;
+        config.max_eth_tx_data_size = 550_000;
+        if config.pubdata_sending_mode == PubdataSendingMode::Blobs
+            || config.pubdata_sending_mode == PubdataSendingMode::Calldata
+        {
+            config.pubdata_sending_mode = PubdataSendingMode::RelayedL2Calldata
+        }
+    }
```

Comment:
let's also add comment on why this is needed and issue a info or warn log that these values were modified